src/BoundaryConditions/field_boundary_conditions.jl

```diff
@@ -127,14 +124,13 @@ and the topology in the boundary-normal direction is used:
  - `ImpenetrableBoundaryCondition` for `Bounded` directions and `Face`-located fields
  - `nothing` for `Flat` directions and/or `Nothing`-located fields
 """
-FieldBoundaryConditions(default_bounded_bc::BoundaryCondition = NoFluxBoundaryCondition();
-                        west = DefaultBoundaryCondition(default_bounded_bc),
-                        east = DefaultBoundaryCondition(default_bounded_bc),
-                        south = DefaultBoundaryCondition(default_bounded_bc),
-                        north = DefaultBoundaryCondition(default_bounded_bc),
-                        bottom = DefaultBoundaryCondition(default_bounded_bc),
-                        top = DefaultBoundaryCondition(default_bounded_bc),
-                        immersed = DefaultBoundaryCondition(default_bounded_bc)) =
+FieldBoundaryConditions(; west = DefaultBoundaryCondition(),
+                          east = DefaultBoundaryCondition(),
+                          south = DefaultBoundaryCondition(),
+                          north = DefaultBoundaryCondition(),
+                          bottom = DefaultBoundaryCondition(),
+                          top = DefaultBoundaryCondition(),
+                          immersed = DefaultBoundaryCondition()) =
```

Comment:
I feel this PR removes a nice feature that I designed for changing the default boundary condition. Note that this code is meant to be applicable to a wide class of models --- not just `NonhydrostaticModel` and `HydrostaticFreeSurfaceModel`.

Reply:
Ok, I can keep the default in the `DefaultBoundaryCondition`.  I guess having it or not is not really the scope of this PR